src/App.jsx

```diff
@@ -1,17 +1,38 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messages from './data/messages.json';
 
 const App = () => {
+  const [entries, setEntries] = useState(
+    messages.map((entry) => ({...entry}))
+  );
+
+  const toggleLike = (id) => {
+    const updatedEntries = entries.map((entry) =>
+      entry.id === id ? { ...entry, liked: !entry.liked } : entry
+    );
+    setEntries(updatedEntries);
+  };
+
+  const totalLikes = entries.filter((entry) => entry.liked).length;
```

Comment:
You could also do this with the[`.reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) method like so:

```javascript
const  totalLikes  =  chatMessages.reduce((count,  message)  =>  count  + (message.liked ? 1 : 0),  0);
```